faster_whisper/transcribe.py

```diff
@@ -587,7 +588,7 @@ def __init__(
         device: str = "auto",
         device_index: Union[int, List[int]] = 0,
         compute_type: str = "default",
-        cpu_threads: int = 16,
+        cpu_threads: int = multiprocessing.cpu_count() // 2,
```

Comment:
This will not be correct on Intel processors that use E and P cores, for example intel i7 12700k has 12 cores and 20 threads, wouldn't it be better to set it to 0 to use the system default value? as even if `cpu_count//2` is correct, this is the number of threads that will be used per process, and if `num_workers` was set to 2 or higher, each worker is going to request the total number of threads to itself which will result in a lot of context switching

Reply:
> This will not be correct on Intel processors that use E and P cores, for example intel i7 12700k has 12 cores and 20 threads, wouldn't it be better to set it to 0 to use the system default value? as even if `cpu_count//2` is correct, this is the number of threads that will be used per process, and if `num_workers` was set to 2 or higher, each worker is going to request the total number of threads to itself which will result in a lot of context switching.
> 
> Refer to `intra_thread` history in CT2 [Changelog](https://github.com/OpenNMT/CTranslate2/blob/master/CHANGELOG.md) and [Performance Tips](https://opennmt.net/CTranslate2/performance.html) for more information

To account for this you could take function in my last message and divide whatever number you want to use by the number of workers, could you not...thus always leaving four threads/logical cores for a user's background tasks on their computer?

Reply:
I'd rather use the CT2 default value which will suit most users, power users always have the option to tune this value to their needs, as you mentioned, each use case will have a best value, there is no one value fits all here

Reply:
Reverting to the original '0' seems best for widest compatibility